File: src/main/java/com/github/nicogrimm/Richtung.java
package com.github.nicogrimm;

public enum Richtung {
    Oben, Rechts, Unten, Links;

    public static Richtung fromChar(char c) {
        return switch (c) {
            case 'w' -> Oben;
            case 'd' -> Rechts;
            case 's' -> Unten;
            case 'a' -> Links;
            default -> null;
        };
    }
}

File: src/main/java/com/github/nicogrimm/Richtung.kt
package com.github.nicogrimm

enum class Richtung {
    Oben, Rechts, Unten, Links;

    companion object {
        fun fromChar(c: Char): Richtung? {
            return when (c) {
                'w' -> Oben
                'd' -> Rechts
                's' -> Unten
                'a' -> Links
                else -> null
            }
        }
    }
}
